python/lsst/meas/base/forcedMeasurement.py

```diff
@@ -250,45 +250,35 @@ def run(self, exposure, refCat, refWcs, idFactory=None, exposureId=None, beginOr
                                  executionOrder >= endOrder are not executed. None for no limit.
         @param[in] allowApCorr  allow application of aperture correction?
 
-        @return SourceCatalog containing measurement results.
-
-        Delegates creating the initial empty SourceCatalog to generateSources(), then fills it.
+        Fills the initial empty SourceCatalog with forced measurement results.
         """
-        # First create a refList from the original which excludes children when a member
-        # of the parent chain is not within the list.  This can occur at boundaries when
-        # the parent is outside and one of the children is within.  Currently, the parent
-        # chain is always only one deep, but just in case, this code removes any reference
-        # when the parent chain to its topmost parent is broken
-
-        # Construct a footprints dict which does not contain the footprint, just the parentID
-        # We will add the footprint from the transformed sources from generateSources later.
-        footprints = {ref.getId(): ref.getParent() for ref in refCat}
-        refList = list()
+        # First check that the reference catalog does not contain any children for which
+        # any member of their parent chain is not within the list.  This can occur at
+        # boundaries when the parent is outside and one of the children is within.
+        # Currently, the parent chain is always only one deep, but just in case, this
+        # code checks for any case where when the parent chain to a child's topmost
+        # parent is broken and raises an exception if it occurs.
+        #
+        # NOTE: this culling should to be accounted for at the reference catalog creation
+        # stage (of which there are several) but is not currently implemented.
```

Comment:
Also, instead of saying the culling is not implemented at the reference catalog stage, I think it's probably more accurate to say the code merely checks that this precondition is satisfied by whatever reference catalog provider is being paired with this code.  We do in fact have reference catalog creation code that guarantees these conditions are satisfied under the conditions in which they are run.